pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotLogicalTableResource.java

```diff
@@ -0,0 +1,256 @@
+  }
+
+  @PUT
+  @Produces(MediaType.APPLICATION_JSON)
+  @Consumes(MediaType.APPLICATION_JSON)
+  @Path("/logicalTables/{tableName}")
+  @Authorize(targetType = TargetType.TABLE, paramName = "tableName", action = Actions.Table.UPDATE_TABLE_CONFIG)
+  @Authenticate(AccessType.UPDATE)
+  @ApiOperation(value = "Update a logical table", notes = "Updates a logical table")
+  @ApiResponses(value = {
+      @ApiResponse(code = 200, message = "Successfully updated schema"), @ApiResponse(code = 404, message = "Schema "
+      + "not found"), @ApiResponse(code = 400, message = "Missing or invalid request body"), @ApiResponse(code = 500,
+      message = "Internal error")
+  })
+  public SuccessResponse updateLogicalTable(
+      @ApiParam(value = "Name of the logical table", required = true) @PathParam("tableName") String tableName,
+      @Context HttpHeaders headers, String logicalTableJsonString) {
+    Pair<LogicalTable, Map<String, Object>> logicalTableAndUnrecognizedProps =
+        getLogicalAndUnrecognizedPropertiesFromJson(logicalTableJsonString);
+    LogicalTable logicalTable = logicalTableAndUnrecognizedProps.getLeft();
+
+    validateLogicalTableName(logicalTable);
+    Preconditions.checkArgument(logicalTable.getTableName().equals(tableName),
+        "Logical table name in the request body should match the table name in the URL");
+
+    tableName = DatabaseUtils.translateTableName(tableName, headers);
+    logicalTable.setTableName(tableName);
+
+    SuccessResponse successResponse = updateLogicalTable(tableName, logicalTable);
+    return new ConfigSuccessResponse(successResponse.getStatus(), logicalTableAndUnrecognizedProps.getRight());
+  }
+
+  @DELETE
+  @Produces(MediaType.APPLICATION_JSON)
+  @Path("/logicalTables/{tableName}")
+  @Authorize(targetType = TargetType.TABLE, paramName = "tableName", action = Actions.Table.DELETE_TABLE)
+  @Authenticate(AccessType.DELETE)
+  @ApiOperation(value = "Delete a logical table", notes = "Deletes a logical table by name")
+  @ApiResponses(value = {
+      @ApiResponse(code = 200, message = "Successfully deleted logical table"), @ApiResponse(code = 404, message =
+      "Logical table not found"), @ApiResponse(code = 500, message = "Error deleting logical table")
+  })
+  public SuccessResponse deleteLogicalTable(
+      @ApiParam(value = "Logical table name", required = true) @PathParam("tableName") String tableName,
+      @Context HttpHeaders headers) {
+    tableName = DatabaseUtils.translateTableName(tableName, headers);
+    if (_pinotHelixResourceManager.deleteLogicalTable(tableName)) {
+      return new SuccessResponse(tableName + " logical table successfully deleted.");
+    } else {
+      throw new ControllerApplicationException(LOGGER, "Failed to delete logical table",
+          Response.Status.INTERNAL_SERVER_ERROR);
+    }
+  }
+
+  private void validateLogicalTableName(LogicalTable logicalTable) {
```

Comment:
This method and `PinotHelixResourceManager.validateLogicalTable` can be merged. I prefer merging the other function to this one.